### ISSUE TITLE
Create an Admin set of Groups

### ISSUE THREAD

**Issue by sgoggins** [chaoss/augur#2276]
Basically, we need a way for an admin user to build repo collections that are commonly used in practice, and then shared by all instance users. They could be hidden for everyone but the admin user in our frontend, but 8Knot needs these repo collections, which they can make visible in their interface because people use these common collections. 

![Image](https://user-images.githubusercontent.com/379847/229637092-400c28ed-d59a-4c95-9033-4d7a6ead575d.png)

**Comment by sgoggins** [comment]
@JamesKunstle : On the redirect button back to 8Knot, the team recalls not wanting to have a back button, and instead having 8Knot popup the Augur window so it can be closed. If we do a back button, we need to: 
1. track the return url
2. distinguish between an instance connected to 8knot and what that isn't...

**Comment by sgoggins** [comment]
@JamesKunstle and @cdolfi : 
We are discussing this and we think the best approach is to use the user that links Augur to 8knot as the user that creates the "Global Groups". For example, in my demo the username was berkeley ... 

If I'm logged in as "Bob", then I would have user groups available for : 

bob_ 
and ALL users would see the groups created by:  
berkeley_ (So, the "not logged in user" would also see the berkeley_ repos ... 

WHILE MY USER NAME IS BERKELEY, you would probably want to create a username for the connection like "project_" or "product_" ... 

So, there'd be one more endpoint 8knot would need to hit, something like this: 

@app.route(f"/{AUGUR_API_VERSION}/universal/groups/names", methods=['GET', 'POST'])

We should probably discuss this a bit further.

**Comment by sgoggins** [comment]
James high level overview: 
1. Lets say you are somebody whose person profile is todd, and you are in charge of a SIG at CNCF (tons of repos) and you want to be able to always look at some subset, and share that subset across any user who has access to your augur instance. 
2. So, perhaps some type of user is an "admin" conceptually and can create this list. 
3. If you create an Augur instance, you can create a top level user group that every other user can access.

**Comment by sgoggins** [comment]
Do we want the public facing groups visible on Augur's front end AND 8Knot's front end? Or just 8knot? 
- Ideally, Augur will have an annotation of groups in the same way as "favorite" is annotated as part of the JSON we get, such that by default, if I ask for the user groups of an individual, they will always be given some set of Admin groups and ALSO their user groups. 
- 8Knot wants Default Groups (Admin created) + a User's set of created of groups.

**Comment by sgoggins** [comment]
1. Add an endpoint to get an application user's groups (application user is the Augur User who created the client id and client secret used to connect to 8Knot, or another application). 
2. Add an endpoint to get the username for who created the application.

**Comment by sgoggins** [comment]
@JamesKunstle : These all follow the same pattern you are using already on existing endpoints. For example:
```
result = requests.post(self.user_groups_endpoint, headers=headers, params=params)
```

A. /api/unstable/application - This gives you information about the application, the user its connected to, and how many users are logged into it.  This lists only the application that the requests came from.

B. /api/unstable/application/group/repos - This gives you all of the repos in the specified group, which belongs to the application user (user who created the application in our admin panel)

C. /api/unstable/application/group/repos/count - Return the number of repos in the group. 

D. /api/unstable/application/groups/names - This lists all the application user's group names. 

**E. /api/unstable/application/groups/repos - This gives you a list of all the application user's groups, and all the repos that are in those groups. It follows the same structure as the user/groups endpoint you already use.**

**Comment by sgoggins** [comment]
Closed by #2311

**Comment by JamesKunstle** [comment]
@sgoggins This thread didn't make it into my inbox until now. I was seeing github comments but I thought that was for something else that I track, I'm sorry for not responding here sooner. I'll be trying this out either later today or tomorrow. Thank you all @IsaacMilarky @Ulincsys @ABrain7710 for your help with this feature, very very appreciated.
